realm-gradle-plugin/src/main/groovy/io/realm/gradle/Realm.groovy

```diff
@@ -42,30 +41,17 @@ class Realm implements Plugin<Project> {
 
         def syncEnabledDefault = false
         def dependencyConfigurationName = getDependencyConfigurationName(project)
-        def usesAptPlugin = project.plugins.findPlugin('com.neenbedankt.android-apt') != null
```

Comment:
While I think we can remove support for this, we cannot do it in a patch release as it might be a breaking change, but if you direct this PR towards the branch `next-major` instead then it should be fine 👍

Reply:
(although technically AS 3.0 cannot run android-apt anymore, but maybe some legacy projects still use it with older version of IDE)

Reply:
@cmelchior  I have changed the branch to next-major :)